posts/2023-09-05-Increasing-Apple-Version-Requirements.md

```diff
@@ -0,0 +1,73 @@
+
+---
+layout: post
+title: "Increasing the minimum supported Apple platform versions"
+author: BlackHoleFox
+description: "Modernizing and improving Apple platform support for Rust"
+---
+
+As of Rust <CURRENT_RUST_VERSION/> (to be released on <DATE/>), the minimum version of Apple's platforms (iOS, macOS, and tvOS) that the Rust toolchain supports will be [increased](https://github.com/rust-lang/rust/pull/104385) to newer minimums. These changes affect both the Rust compiler itself (`rustc`), other host tooling, and most importantly, the standard library and any binaries produced that use it. With these changes in place, any binaries produced will stop loading on older versions or exhibit other, unspecified, behavior.
+
+The new minimum versions are now:
+- macOS: 10.12 Sierra (First released 2016)
+- iOS: 10 (First released 2016)
+- tvOS: 10 (First released 2016)
+
+If your application does not target or support macOS 10.7-10.11, or iOS 7-9 already these changes most likely do not affect you.
+
+# Affected targets
+
+The following contains each affected target, and the comprehensive effects on it:
+- `x86_64-apple-darwin` (Minimum OS raised)
+- `aarch64-apple-ios` (Minimum OS raised)
+- `aarch64-apple-ios-sim` (Minimum iOS *and macOS* version raised.)
+- `x86_64-apple-ios` (Minimum iOS *and macOS* version raised. This is also simulator target.)
+- `aarch64-apple-tvos` (Minimum OS raised)
+- `armv7-apple-ios` (Target removed. This oldest iOS 10-compatible device uses ARMv7s)
+- `armv7s-apple-ios` (Minimum OS raised)
+- `i386-apple-ios` (Minimum OS raised)
+- `i686-apple-darwin` (Minimum OS raised)
+- `x86_64-apple-tvos` (Minimum tvOS *and macOS* version raised. This is also a simulator target.)
+
+From these changes, only one target has been removed entirely: `armv7-apple-ios`. It was a tier 3 target.
+
+Note that Mac Catalyst and M1/M2 (`aarch64`) Mac targets are not affected, as their minimum OS version already has a higher baseline. Refer to the [Platform Support Guide](https://doc.rust-lang.org/rustc/platform-support.html) for more information.
+
+# Affected systems
+
+These changes remove support for multiple older iDevices and many more Mac systems. Thanks to `@madsmtm` for [compiling the list](https://github.com/rust-lang/rust/pull/104385#issuecomment-1317830217).
+
+As of this update, the following devices are no longer supported by the latest Rust toolchain:
+
+### iOS
+
+-   iPhone 4S (Released in 2011)
+-   iPad 2 (Released in 2011)
+-   iPad, 3rd generation (Released in 2012)
+-   iPad Mini, 1st generation (Released in 2012)
+-   iPod Touch, 5th generation (Released in 2012)
+
+### macOS
+
+A total of 27 Mac computers, released between 2007 and 2009, are no longer supported.
```

Comment:
```suggestion
A total of 27 Mac models, released between 2007 and 2009, are no longer supported.
```
> A total of 27 Mac computers

makes it sound a bit like only 27 physical devices are actually affected. I'm not sure whether model is the best word here though.

Reply:
I own all 27 of those physical devices /s

Went for "system models" to ideally make it non-confusable.